### DISCUSSION TITLE
Compile-time flags for compiling CPU vs GPU version of port

### DISCUSSION THREAD

**Discussion by marcodiiga** [microsoft/vcpkg#32692] [Q&A] [answered]
A user had some issues getting a CPU-only version of the physx port to work (https://github.com/marcodiiga/hello_physx_vcpkg/issues/1). I believe this should be a compile-time flag when installing the port and compiling it, is there a standard way to pass extra-flags to vcpkg install? How are compile-time-affecting flags handled usually?

I found [this experimental](https://learn.microsoft.com/en-us/vcpkg/commands/install#feature) flag but I'm not sure if it's the right way to go about this.

**Comment by Osyotr** [comment, marked as answer]
You need to define `DISABLE_CUDA_PHYSX` while *building* physx (and probably while building *your project*, too).
To build physx with this define, add this to your [custom triplet](https://learn.microsoft.com/en-us/vcpkg/users/examples/overlay-triplets-linux-dynamic):
```cmake
list(APPEND VCPKG_C_FLAGS -DDISABLE_CUDA_PHYSX)
list(APPEND VCPKG_CXX_FLAGS -DDISABLE_CUDA_PHYSX)
```
Alternatively you can create an [overlay port](https://learn.microsoft.com/en-us/vcpkg/commands/common-options#overlay-ports) and patch physx (uncomment this line https://github.com/NVIDIA-Omniverse/PhysX/blob/8f6cf9cfe89f2bedafbf7788a72cb04f11c31e1f/physx/include/foundation/PxPreprocessor.h#L509)

PS: Vote for #22216 if you want an easier way to achieve that! I.e. `vcpkg install physx[gpu=off]`
